File: 프로그래머스/1/388351. 유연근무제/유연근무제.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> schedules, vector<vector<int>> timelogs, int startday) {
    int answer = 0;
    // 주말은 3000으로 만들기
    for(vector<int>& person : timelogs)
    {
        if (startday == 7)
        {
            person[7-startday] = 0;
            person[6] = 0;
        }
        else
        {
            person[7-startday] = 0;
            person[6-startday] = 0;   
        }
    }
    // schedules + 10 보다 작으면 answer 에 +1
    int seq = 0;
    bool flag = false;
    for(vector<int>& person : timelogs)
    {   
        flag = false;
        // 50분 이면 다음시간으로 넘겨주기
        int temp_time = schedules[seq];
        if(temp_time%100 >= 50)
        {
            temp_time += 50;
        }
        else
        {
            temp_time += 10;
        }
        int deadline = temp_time;
        //cout << "deadline:" << deadline << '\n';
        for(int& time : person)
        {   
            //cout << "time:" << time << ' ';
            if (time > deadline)
            {
                flag = true;
                continue;
            }
        }
        if (flag == false)
        {
            answer += 1;   
        }
        seq += 1;
    }
    
    
    return answer;
}

File: 프로그래머스/1/388351. 유연근무제/유연근무제.py
def solution(schedules, timelogs, startday):
    answer = len(schedules)
    # 토,일요일 index설정
    sat = 7-startday
    sun = 7-startday+1
    if sun <= 0:
        sun += 7
    if sat <= 0:
        sat += 7
    # 토,일요일 무조건 통과로 취급 -> 02시로 바꾸기
    for timelog in timelogs:
        timelog[sat-1] = 200
        timelog[sun-1] = 200
    # 시간을 분으로 바꾸기
    # 1. timelogs
    for timelog in timelogs:
        for i in range(len(timelog)):
            timelog[i] = (timelog[i]//100) * 60 + timelog[i]%100
    # 2. schedules
    for i in range(len(schedules)):
        schedules[i] = (schedules[i]//100) * 60 + schedules[i]%100
    # 검사
    for i in range(len(schedules)):
        schedule = schedules[i]
        timelog = timelogs[i]
        for j in range(len(timelog)):
            if schedule+10 < timelog[j]:
                answer -= 1
                break
    return answer